pkg/apis/openstack/validation/controlplane.go

```diff
@@ -138,13 +138,18 @@ func validateLoadBalancerClassesConstraints(floatingPools []api.FloatingPool, sh
 	return allErrs
 }
 
-func validateStorage(storage *api.Storage, shareNetwork *api.ShareNetwork, fldPath *field.Path) field.ErrorList {
+func validateStorage(storage *api.Storage, networks api.Networks, fldPath *field.Path) field.ErrorList {
 	var allErrs field.ErrorList
 	if storage == nil || storage.CSIManila == nil || !storage.CSIManila.Enabled {
 		return allErrs
 	}
-	if shareNetwork == nil || !shareNetwork.Enabled {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("csiManila", "enabled"), storage.CSIManila.Enabled, "share network must be created if CSI manila driver is enabled"))
+	// for an existing subnet we do not need to validate the storage since we can only do that in runtime.
+	if networks.SubnetID != nil {
+		return allErrs
 	}
-	return allErrs
+	if networks.ShareNetwork != nil && networks.ShareNetwork.Enabled {
+		return allErrs
+
+	}
+	return append(allErrs, field.Invalid(fldPath.Child("csiManila", "enabled"), storage.CSIManila.Enabled, "share network must be created if CSI manila driver is enabled"))
```

Comment:
Doing it like this will not necessarily detect invalid configurations, e.g. `networks.ShareNetwork.Enabled` being set but `ShareNetwork` not being given while `networks.SubnetID` is given.
I suspect that won't cause issues in the actual code. Still, not sure if short-cutting in validation is the way to go, I'd prefer it to be exhaustive and explicit.